include/madwf_ml.h

```diff
@@ -0,0 +1,372 @@
+#pragma once
```

Comment:
This is a lot of implementation code in a header.  Can most of this be moved to a .cpp file?  Moreover, if we were to use MADWF in other solvers, we'd end up with unnecessary bloat.

Reply:
https://github.com/lattice/quda/pull/1061/commits/96926239a741adb4fa5c8bab17dc6714e094af3b